### PR TITLE
gtkgui/userbrowse.py: remember selection after refresh

### PR DESCRIPTION
**Pull request by slook**
If the file and/or folder still exists in the refreshed shares, then it was frustrating that the selection is lost after doing a refresh operation, because finding the desired item again can be difficult if the shares are large, as there are no location history features.

+ Added: Return to the currently selected location (file and/or folder) after doing a Refresh
+ Changed: Add a trailing slash onto to the end of folder path copy to clipboard string
+ Fixed: Grab the focus onto the relevant list view widget when entering a URL even if the tab is already open

The clipboard copy code is slightly repurposed for this. The new getter functions are designed to be useful for other purposes too, such as for storing location history, status text, tooltips, etc.

If the selected file doesn't exist anymore, then the folder is selected at least. Only a single file selection is recalled (upper-most).

### CODE REVIEW

File: pynicotine/gtkgui/userbrowse.py [hunk 1]
```diff
@@ -489,8 +489,8 @@ def browse_queued_path(self):
         if not iterator:
             return
 
-        # Scroll to the requested file
-        self.file_list_view.select_row(iterator)
+        # Scroll to the requested file and grab focus to it
+        self.file_list_view.select_row(iterator, should_expand=True)
```
Comment (mathiascode) [inline pynicotine/gtkgui/userbrowse.py:493]: This feels strange, since we don't have expanders in the file list.

Comment (slook) [reply pynicotine/gtkgui/userbrowse.py:493]: I know, this is my least liked element of this PR.

Functionally it works perfectly, but it feels like a hack.

Suggest perhaps renaming/reworking the method in treeview.py to seperate:
- `should_scroll`
- `should_expand`
- `should_focus`

I cannot think of a scenario where we want to expand but not focus, but I could be mistaken.

Comment (mathiascode) [reply pynicotine/gtkgui/userbrowse.py]: I think you could just call grab_focus() when should_focus is True. There shouldn't be any situations where we don't want to focus the widget when scrolling to a row.

Comment (slook) [reply pynicotine/gtkgui/userbrowse.py]: No, we can't do that because it breaks the search next match behaviour.

Comment (mathiascode) [reply pynicotine/gtkgui/userbrowse.py]: That's a shame. I'll have to take a closer look, since the select_row() method is becoming dangerously large.

Comment (slook) [reply pynicotine/gtkgui/userbrowse.py]: I reverted the changes to treeview.py in favour of grabbing the focus in browse_queued_path()

File: pynicotine/gtkgui/userbrowse.py [hunk 2]
```diff
@@ -1044,25 +1047,17 @@ def on_file_properties(self, _action, _state, all_files=False):
...
 
     def on_copy_file_path(self, *_args):
-
-        if self.selected_folder is None or not self.selected_files:
-            return
-
-        text = "\\".join([self.selected_folder, next(iter(self.selected_files))])
-        copy_text(text)
+        file_path = self.get_selected_file_path()
+        copy_text(f"{self.user}\\{file_path}")
```
Comment (mathiascode) [inline pynicotine/gtkgui/userbrowse.py:1051]: Changes to paths should not be included in this PR

Comment (slook) [reply pynicotine/gtkgui/userbrowse.py]: reverted.
